rustls/src/client/hs.rs

```diff
@@ -271,6 +271,47 @@ fn emit_client_hello_for_retry(
     // Do we have a SessionID or ticket cached for this host?
     let tls13_session = prepare_resumption(&input.resuming, &mut exts, suite, cx, config);
 
+    // Extensions MAY be randomised
+    // PSK message must be last, if it is we cannot move it
+    let randomisable_extensions = exts.len()
+        - if let Some(&ClientExtension::PresharedKey(..)) = exts.last() {
```

Comment:
Style nit: I think this would look be better if you use a `match` instead of `if let`.

Despite the parent issue, IIRC we do use US English spelling through the code base, so let's call it "randomized" and "randomizable"?

Reply:
Done!